apps/server/src/modules/school/api/test/school-patch.api.spec.ts

```diff
@@ -0,0 +1,327 @@
+	});
+
+	beforeEach(async () => {
+		await cleanupCollections(em);
+	});
+
+	afterAll(async () => {
+		await app.close();
+	});
+
+	describe('PATCH /:id', () => {
+		describe('when user is not logged in', () => {
+			it('should return 401', async () => {
+				const response = await testApiClient.patch('id').send({
+					name: 'new name',
+				});
+
+				expect(response.status).toEqual(HttpStatus.UNAUTHORIZED);
+			});
+		});
+
+		describe('when user is logged in', () => {
+			describe('when user is an admin', () => {
+				describe('when request is not valid', () => {
+					const setup = async () => {
+						const school = schoolEntityFactory.build();
+						const { adminAccount, adminUser } = UserAndAccountTestFactory.buildAdmin({ school });
+
+						await em.persistAndFlush([adminAccount, adminUser, school]);
+						em.clear();
+
+						const loggedInClient = await testApiClient.login(adminAccount);
+
+						return { loggedInClient, school };
+					};
+
+					describe('when id in params is not a mongo id', () => {
+						it('should return 400', async () => {
+							const { loggedInClient } = await setup();
+
+							const response = await loggedInClient.get(`id/123`);
+
+							expect(response.status).toEqual(HttpStatus.BAD_REQUEST);
+							// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+							expect(response.body.validationErrors).toEqual([
+								{ errors: ['schoolId must be a mongodb id'], field: ['schoolId'] },
+							]);
+						});
+					});
+
+					describe('when requested school is not found', () => {
+						it('should return 404', async () => {
+							const { loggedInClient } = await setup();
+
+							const response = await loggedInClient.patch(new ObjectId().toHexString()).send({
+								name: 'new name',
+							});
+
+							expect(response.status).toEqual(HttpStatus.NOT_FOUND);
+						});
+					});
+
+					describe('when FileStorageType param is not valid', () => {
+						it('should return 404', async () => {
+							const { loggedInClient, school } = await setup();
+
+							const response = await loggedInClient.patch(school.id).send({
+								fileStorageType: 'invalid',
+							});
+
+							expect(response.status).toEqual(HttpStatus.BAD_REQUEST);
+							// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+							expect(response.body.validationErrors).toEqual([
+								{ errors: ['fileStorageType must be one of the following values: awsS3'], field: ['fileStorageType'] },
+							]);
+						});
+					});
+
+					describe('when language param is not valid', () => {
+						it('should return 404', async () => {
+							const { loggedInClient, school } = await setup();
+
+							const response = await loggedInClient.patch(school.id).send({
+								language: 'invalid',
+							});
+
+							expect(response.status).toEqual(HttpStatus.BAD_REQUEST);
+							// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+							expect(response.body.validationErrors).toEqual([
```

Comment:
expect(response.body)..toEqual({ validationErrors: [ ... ] 
to avoid the eslint-disable

Same for each other place